src/constraints.jl

```diff
@@ -305,6 +305,23 @@ function is_valid(model::Model, con_ref::ConstraintRef{<:AbstractModel})
     )
 end
 
+"""
+    is_valid(model::Model, c::ConstraintRef{Model,NonlinearConstraintIndex})
+
+Return `true` if `constraint_ref` refers to a valid nonlinear constraint in
+`model`.
+"""
+function is_valid(
+    model::Model,
+    c::ConstraintRef{Model,NonlinearConstraintIndex},
+)
+    if model !== c.model
+        return false
+    end
+    _init_NLP(model)
+    return 1 <= c.index.value <= num_nl_constraints(model)
+end
```

Comment:
Ah. I missed this on first glance. It needs to go in the `src/nlp.jl` file after `NonlinearConstraintIndex` is defined.

Reply:
yeah, I'm moving it over there. Noticed it after precompilation errors came up